File: CroceVerdePlus/app/src/main/java/com/example/croceverdeplus/Milite.kt
package com.example.croceverdeplus

data class Milite(
    val username: String? = null,
    val password: String? = null,
    var nome: String? = null,
    var cognome: String? = null,
    var dataDiNascita: String? = null,
    var residenza: String? = null,
    val ruolo: String? = null,
    var cognomeNomeSpinner: String? = null,
    var grado118prima: Boolean? = null,
    var grado118seconda: Boolean? = null,
    var grado118terza: Boolean? = null,
    var gradoh24prima: Boolean? = null,
    var gradoh24seconda: Boolean? = null,
    var gradoh24terza: Boolean? = null,
    var oreTurno118prima: Int? = null,
    var oreTurno118seconda: Int? = null,
    var oreTurno118terza: Int? = null,
    var oreTurnoh24prima: Int? = null,
    var oreTurnoh24seconda: Int? = null,
    var oreTurnoh24terza: Int? = null) {


    constructor(nome: String?, cognome: String?,
        dataDiNascita: String?, residenza: String?,
                grado118prima: Boolean?, grado118seconda: Boolean?,
                grado118terza: Boolean?, gradoh24prima: Boolean?,
                gradoh24seconda: Boolean?, gradoh24terza: Boolean?) : this() {
        this.nome = nome
        this.cognome = cognome
        this.dataDiNascita = dataDiNascita
        this.residenza = residenza
        this.grado118prima = grado118prima
        this.grado118seconda = grado118seconda
        this.grado118terza = grado118terza
        this.gradoh24prima = gradoh24prima
        this.gradoh24seconda = gradoh24seconda
        this.gradoh24terza = gradoh24terza

    }
}

File: CroceVerdePlusFlutter/lib/milite.dart
class Milite {
  final String? nome;
  final String? cognome;
  final String? dataDiNascita;
  final String? residenza;
  final String? cognomeNomeSpinner;
  final String? username;
  final String? password;
  final bool? grado118prima;
  final bool? grado118seconda;
  final bool? grado118terza;
  final bool? gradoh24prima;
  final bool? gradoh24seconda;
  final bool? gradoh24terza;
  final int? oreTurno118prima;
  final int? oreTurno118seconda;
  final int? oreTurno118terza;
  final int? oreTurnoh24prima;
  final int? oreTurnoh24seconda;
  final int? oreTurnoh24terza;


  Milite({
    required this.nome,
    required this.cognome,
    required this.dataDiNascita,
    required this.residenza,
    this.cognomeNomeSpinner,
    this.username,
    this.password,
    this.grado118prima = false,
    this.grado118seconda = false,
    this.grado118terza = false,
    this.gradoh24prima = false,
    this.gradoh24seconda = false,
    this.gradoh24terza = false,
    this.oreTurno118prima = 0,
    this.oreTurno118seconda = 0,
    this.oreTurno118terza = 0,
    this.oreTurnoh24prima = 0,
    this.oreTurnoh24seconda = 0,
    this.oreTurnoh24terza = 0,
  });



  Map<String, dynamic> toMap() {
    return {
      'nome': nome,
      'cognome': cognome,
      'dataDiNascita': dataDiNascita,
      'residenza': residenza,
      'cognomeNomeSpinner': cognomeNomeSpinner,
      'username': username,
      'password': password,
      'grado118prima': grado118prima,
      'grado118seconda': grado118seconda,
      'grado118terza': grado118terza,
      'gradoh24prima': gradoh24prima,
      'gradoh24seconda': gradoh24seconda,
      'gradoh24terza': gradoh24terza,
      'oreTurno118prima': oreTurno118prima,
      'oreTurno118seconda': oreTurno118seconda,
      'oreTurno118terza': oreTurno118terza,
      'oreTurnoh24prima': oreTurnoh24prima,
      'oreTurnoh24seconda': oreTurnoh24seconda,
      'oreTurnoh24terza': oreTurnoh24terza,
    };
  }
}

